fastembed/common/model_management.py

```diff
@@ -120,23 +121,52 @@ def download_files_from_huggingface(
             "special_tokens_map.json",
             "preprocessor_config.json",
         ]
-        if extra_patterns is not None:
-            allow_patterns.extend(extra_patterns)
+
+        model_file = next((file for file in extra_patterns if file.endswith(".onnx")), "")
+        allow_patterns.extend(extra_patterns)
 
         snapshot_dir = Path(cache_dir) / f"models--{hf_source_repo.replace('/', '--')}"
-        is_cached = snapshot_dir.exists()
+        metadata_file = snapshot_dir / "files_metadata.json"
+
+        def _verify_files_from_metadata(model_dir: Path, stored_metadata: dict[str, Any]) -> bool:
+            for rel_path, meta in stored_metadata.items():
+                file_path = model_dir / rel_path
+                if not file_path.exists() or file_path.stat().st_size != meta["size"]:
+                    return False
+            return True
+
+        def _save_file_metadata(model_dir: Path) -> None:
+            metadata = {}
+            for file_path in model_dir.rglob("*"):
+                if file_path.is_file() and file_path.name != "files_metadata.json":
+                    rel_path = str(file_path.relative_to(model_dir))
+                    metadata[rel_path] = {
+                        "size": file_path.stat().st_size,
+                    }
+
+            metadata_file = model_dir / "files_metadata.json"
+            metadata_file.write_text(json.dumps(metadata))
 
-        if is_cached:
-            disable_progress_bars()
+        if snapshot_dir.exists() and metadata_file.exists():
```

Comment:
if version on hf is different from the one we have locally, then we will hide the progress bar and then we will silently  download the updated files
I think we could make a corresponding call to HfApi and check revision and commit hash

Reply:
I think it's safe to make this call if local_files_only != True

Reply:
iirc in snapshot_download they just pull the whole repo info and compare revision's commit hahs